### PR TITLE
MongoDBStitch library implementation

### PR DESCRIPTION
**Pull request by ekaterina-nbl**

### CODE REVIEW

File: Examples/README.md [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+## DataSender
+
+This example inserts data into preconfigured MongoDB collection using preconfigured MongoDB Stitch named pipeline:
+
+- Data are sent every 10 seconds.
+- Every data record contains:
+  - `"value"` attribute - integer value, converted to string, which starts at 1 and increases by 1 with every record sent. It restarts from 1 everytime when the example is restarted.
+  - `"measureTime"` attribute - integer value, converted to string, which is the time in seconds since the epoch.
+
+![DataSender example](https://imgur.com/fM75b66.png)
```
Comment (edaniels) [inline Examples/README.md]: I'd recommend you store these images locally or in a stable location that is not imgur for longevity purposes.

File: Examples/DataSender.agent.nut [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    }
+}
+
+// RUNTIME
+// ---------------------------------------------------------------------------------
+
+// MONGO DB STITCH CONSTANTS
+// ----------------------------------------------------------
+const MONGO_DB_STITCH_APPLICATION_ID = "<YOUR_STITCH_APPLICATION_ID>";
+const MONGO_DB_STITCH_API_KEY = "<YOUR_STITCH_API_KEY>";
```
Comment (edaniels) [inline Examples/DataSender.agent.nut:92]: [q] Is this kind of practice okay for electricimp? I'm not sure if people would accidentally commit this to github or not or if there is a way to have secure data configuration.

**Review by edaniels — COMMENTED** [review]
Looks fine to me as it currently works with Client API v1. I would hold on merging this until Client API v2 is implemented instead with the new features.

File: Examples/DataSender.agent.nut (hunk 2 above)
Comment (ppetrosh) [reply Examples/DataSender.agent.nut:92]: Hi @edaniels, yes this is a common practice (not ideal but this is what it is). In fact you can use https://github.com/electricimp/builder not to hardcode things in the application code and parameterize it at "build" time (which is the preferred/recommended but more sophisticated way).

File: Examples/README.md (hunk 1 above)
Comment (ppetrosh) [reply Examples/README.md]: all the images are available locally of course. but I agree with you, we should probably use local references instead

**Comment by ppetrosh** [comment]
@edaniels as for the v2 APIs what's the timeframe for getting them out?

**Comment by edaniels** [comment]
Probably next Tuesday

**Comment by edaniels** [comment]
@ekaterina-nbl @alexey-nbl @ppetrosh if you want me to take another look at this once functions are done or run into any issues, just let me know. I'll be around all week/end.

**Comment by alexey-nbl** [comment]
@edaniels @ppetrosh the pull request is updated by the updated lib, readme, examples, tests: moved to v2 API, NamedPipelines replaced by Functions. Please review and integrate.

File: MongoDBStitch.agent.lib.nut [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+        } catch (e) {
+            if (!errType) {
+                errType = MONGO_DB_STITCH_ERROR.MONGO_DB_STITCH_UNEXPECTED_RESPONSE;
+                errDetails = e;
+            }
+        }
+        
+        // try to refresh token in case it is expired
+        if (errType == MONGO_DB_STITCH_ERROR.MONGO_DB_STITCH_REQUEST_FAILED
+            && httpStatus == 401 && _getTableValue(response.body, "errorCode", null) == "InvalidSession"
```
Comment (edaniels) [inline MongoDBStitch.agent.lib.nut]: Use error_code here instead of errorCode

**Review by edaniels — CHANGES_REQUESTED** [review]
Just one thing I noticed

File: Examples/DataSender.agent.nut (hunk 2 above)
Comment (ppetrosh) [reply Examples/DataSender.agent.nut:92]: @edaniels fixed.

File: MongoDBStitch.agent.lib.nut (hunk 3 above)
Comment (ppetrosh) [reply MongoDBStitch.agent.lib.nut]: @edaniels will take care of it, thanks!

**Review by edaniels — APPROVED** [review]

**Review by ppetrosh — APPROVED** [review]

**Pull request merged**